// ngx-ionic-image-viewer/bundles/ngx-ionic-image-viewer.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@ionic/angular'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ngx-ionic-image-viewer', ['exports', '@angular/core', '@ionic/angular', '@angular/common'], factory) :
    (global = global || self, factory(global['ngx-ionic-image-viewer'] = {}, global.ng.core, global['@ionic/angular'], global.ng.common));
}(this, (function (exports, core, angular, common) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __exportStar(m, exports) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var m = typeof Symbol === "function" && o[Symbol.iterator], i = 0;
        if (m) return m.call(o);
        return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgxIonicImageViewerService = /** @class */ (function () {
        function NgxIonicImageViewerService() {
        }
        NgxIonicImageViewerService.decorators = [
            { type: core.Injectable, args: [{
                        providedIn: 'root'
                    },] }
        ];
        /** @nocollapse */
        NgxIonicImageViewerService.ctorParameters = function () { return []; };
        /** @nocollapse */ NgxIonicImageViewerService.ngInjectableDef = core.ɵɵdefineInjectable({ factory: function NgxIonicImageViewerService_Factory() { return new NgxIonicImageViewerService(); }, token: NgxIonicImageViewerService, providedIn: "root" });
        return NgxIonicImageViewerService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var ViewerModalComponent = /** @class */ (function () {
        function ViewerModalComponent(modalController) {
            this.modalController = modalController;
            this.defaultSlideOpts = {
                centeredSlides: 'true',
                zoom: {
                    enabled: true
                }
            };
            this.options = __assign({}, this.defaultSlideOpts);
        }
        /**
         * @return {?}
         */
        ViewerModalComponent.prototype.ngOnInit = /**
         * @return {?}
         */
        function () {
            this.slides.update();
        };
        /**
         * @return {?}
         */
        ViewerModalComponent.prototype.closeModal = /**
         * @return {?}
         */
        function () {
            this.modalController.dismiss();
        };
        ViewerModalComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ion-viewer-modal',
                        template: "<ion-header [ngClass]=\"{ 'no-title': title.length <= 0 }\">\n  <ion-toolbar>\n    <ion-buttons slot=\"primary\">\n      <ion-button (click)=\"closeModal()\"> <ion-icon slot=\"icon-only\" name=\"close\"></ion-icon> </ion-button>\n    </ion-buttons>\n    <ion-title>{{ title }}</ion-title>\n  </ion-toolbar>\n</ion-header>\n\n<ion-content [forceOverscroll]=\"false\" fullscreen=\"true\">\n  <ion-slides [options]=\"options\" #sliderRef>\n    <ion-slide>\n      <div class=\"swiper-zoom-container\">\n        <img alt=\"alt\" [src]=\"src\" height=\"100vh\" />\n      </div>\n    </ion-slide>\n  </ion-slides>\n</ion-content>\n\n<ion-footer [ngClass]=\"{ 'no-text': text.length <= 0 }\">\n  <ion-toolbar class=\"ion-text-center\">\n    <ion-text>{{ text }}</ion-text>\n  </ion-toolbar>\n</ion-footer>\n",
                        styles: ["ion-slides{--height:100%;height:100%}ion-toolbar{--border-style:none;--background:rgba(var(--ion-background-color-rgb, (255, 255, 255)), 0.6);background:rgba(var(--ion-background-color-rgb,255,255,255),.6)}ion-header{position:absolute}ion-header.no-title:after{content:none}ion-header.no-title ion-toolbar{--background:rgba(0, 0, 0, 0);background:rgba(0,0,0,0)}ion-footer{position:absolute;bottom:0}ion-footer.no-text:before{content:none}ion-footer.no-text ion-toolbar{--background:rgba(0, 0, 0, 0);background:rgba(0,0,0,0)}"]
                    }] }
        ];
        /** @nocollapse */
        ViewerModalComponent.ctorParameters = function () { return [
            { type: angular.ModalController }
        ]; };
        ViewerModalComponent.propDecorators = {
            src: [{ type: core.Input }],
            alt: [{ type: core.Input }],
            title: [{ type: core.Input }],
            text: [{ type: core.Input }],
            options: [{ type: core.Input }],
            slides: [{ type: core.ViewChild, args: ['sliderRef', { static: true },] }]
        };
        return ViewerModalComponent;
    }());
    if (false) {
        /** @type {?} */
        ViewerModalComponent.prototype.src;
        /** @type {?} */
        ViewerModalComponent.prototype.alt;
        /** @type {?} */
        ViewerModalComponent.prototype.title;
        /** @type {?} */
        ViewerModalComponent.prototype.text;
        /** @type {?} */
        ViewerModalComponent.prototype.defaultSlideOpts;
        /** @type {?} */
        ViewerModalComponent.prototype.options;
        /** @type {?} */
        ViewerModalComponent.prototype.slides;
        /**
         * @type {?}
         * @private
         */
        ViewerModalComponent.prototype.modalController;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgxIonicImageViewerComponent = /** @class */ (function () {
        function NgxIonicImageViewerComponent(modalController) {
            this.modalController = modalController;
        }
        /**
         * @param {?} src
         * @param {?=} title
         * @param {?=} text
         * @return {?}
         */
        NgxIonicImageViewerComponent.prototype.viewImage = /**
         * @param {?} src
         * @param {?=} title
         * @param {?=} text
         * @return {?}
         */
        function (src, title, text) {
            if (title === void 0) { title = ''; }
            if (text === void 0) { text = ''; }
            return __awaiter(this, void 0, void 0, function () {
                var modal;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, this.modalController.create({
                                component: ViewerModalComponent,
                                componentProps: {
                                    src: src,
                                    title: title,
                                    text: text
                                },
                                cssClass: 'modal-fullscreen',
                                keyboardClose: true,
                                showBackdrop: true
                            })];
                        case 1:
                            modal = _a.sent();
                            return [4 /*yield*/, modal.present()];
                        case 2: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        };
        /**
         * @return {?}
         */
        NgxIonicImageViewerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
        function () { };
        NgxIonicImageViewerComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ion-img-viewer',
                        template: "<ion-img [alt]=\"alt\" [src]=\"src\" (click)=\"viewImage(src, title, text)\"></ion-img>\n",
                        encapsulation: core.ViewEncapsulation.None,
                        styles: ["\n      .modal-fullscreen {\n        --border-radius: 0;\n        border-radius: 0;\n      }\n      .modal-fullscreen .modal-wrapper {\n        position: absolute;\n        top: 0;\n        left: 0;\n        display: block;\n        width: 100%;\n        height: 100%;\n      }\n    "]
                    }] }
        ];
        /** @nocollapse */
        NgxIonicImageViewerComponent.ctorParameters = function () { return [
            { type: angular.ModalController }
        ]; };
        NgxIonicImageViewerComponent.propDecorators = {
            src: [{ type: core.Input }],
            alt: [{ type: core.Input }],
            title: [{ type: core.Input }],
            text: [{ type: core.Input }]
        };
        return NgxIonicImageViewerComponent;
    }());
    if (false) {
        /** @type {?} */
        NgxIonicImageViewerComponent.prototype.src;
        /** @type {?} */
        NgxIonicImageViewerComponent.prototype.alt;
        /** @type {?} */
        NgxIonicImageViewerComponent.prototype.title;
        /** @type {?} */
        NgxIonicImageViewerComponent.prototype.text;
        /** @type {?} */
        NgxIonicImageViewerComponent.prototype.modalController;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgxIonicImageViewerModule = /** @class */ (function () {
        function NgxIonicImageViewerModule() {
        }
        NgxIonicImageViewerModule.decorators = [
            { type: core.NgModule, args: [{
                        declarations: [NgxIonicImageViewerComponent, ViewerModalComponent],
                        imports: [common.CommonModule, angular.IonicModule],
                        entryComponents: [ViewerModalComponent],
                        exports: [NgxIonicImageViewerComponent, ViewerModalComponent]
                    },] }
        ];
        return NgxIonicImageViewerModule;
    }());

    exports.NgxIonicImageViewerComponent = NgxIonicImageViewerComponent;
    exports.NgxIonicImageViewerModule = NgxIonicImageViewerModule;
    exports.NgxIonicImageViewerService = NgxIonicImageViewerService;
    exports.ɵa = ViewerModalComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
